src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java

```diff
@@ -0,0 +1,129 @@
+package thePackmaster.cards.rippack;
+
+import basemod.BaseMod;
+import basemod.helpers.TooltipInfo;
+import com.evacipated.cardcrawl.mod.stslib.patches.HitboxRightClick;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.animations.VFXAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.UIStrings;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import com.megacrit.cardcrawl.vfx.ThoughtBubble;
+import thePackmaster.actions.rippack.DividedFuryAction;
+import thePackmaster.actions.rippack.RipCardAction;
+import thePackmaster.powers.rippack.DividedFuryPower;
+import thePackmaster.vfx.rippack.ShowCardAndRipEffect;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+import static thePackmaster.util.Wiz.atb;
+import static thePackmaster.util.Wiz.att;
+
+public abstract class AbstractRippableCard extends AbstractRipCard {
+    private static final UIStrings uiStrings = CardCrawlGame.languagePack.getUIString(makeID("Rip"));
+    private AbstractGameAction action;
+    protected ArrayList<AbstractCard> rippedParts;
+    private static ArrayList<TooltipInfo> consumableTooltip;
+    public static int cardsRippedThisTurn;
+
+    public AbstractRippableCard(String cardID, int cost, CardType type, CardRarity rarity, CardTarget target) {
+        super(cardID, cost, type, rarity, target);
+    }
+
+    protected void setRippedCards(AbstractRippedArtCard artCard, AbstractRippedTextCard textCard) {
+        rippedParts = new ArrayList<>();
+        if(upgraded) {
+            artCard.upgrade();
+            textCard.upgrade();
+        }
+        rippedParts.add(artCard);
+        rippedParts.add(textCard);
+    }
+
+    public ArrayList<AbstractCard> getRippedParts() {
+        return rippedParts;
+    }
+
+    public void onRightClick() {
+        if(action == null) {
+            if (canRip()) {
+                CardCrawlGame.sound.play("MAP_CLOSE", 0.0F);
+                action = new RipCardAction(this, rippedParts.get(0), rippedParts.get(1));
+                att(action);
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
```

Comment:
I am not a happy camper

Reply:
Without these, the two halves of the card go into your hand before the animation of the rip finishes. I could do what I do elsewhere where I force other actions to wait for one to be done before they start

Reply:
Might be better to use a ForceWaitAction if you're scared that people will circumvent the wait with SuperFastMode. (or just instnaciate the action, set its duration to whatever and then queue it.